### PR TITLE
[1.0_branch] Add Optuna sweeper plugin

### PR DESCRIPTION
**Pull request by toshihikoyanase**
## Motivation

This PR backports the change in  #1132 to the 1.0 branch. It cherry-picked the commit https://github.com/facebookresearch/hydra/commit/4f1018d7e0575cd394020bb668d1a8a3243fa9aa.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebookresearch/hydra/blob/master/CONTRIBUTING.md)?

Yes

## Test Plan

This PR includes the same tests as  #1132.

## Related Issues and PRs

The original PR is  #1132.

### CODE REVIEW

**Comment by omry** [comment]
I am holding off merging it because a followup would be to make an initial release of the plugin, would you like to take care of it?

**Comment by toshihikoyanase** [comment]
@omry
I'm interested in the followup. Could you briefly explain it?

**Comment by omry** [comment]
Sure, have you ever released a pip package?
read about it.

Basically, we should release a package for it. after your initial release add me as a second owner in pypi (user omry).

**Review by omry — CHANGES_REQUESTED** [review]

File: plugins/hydra_optuna_sweeper/setup.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+def get_long_description() -> str:
+
+    readme_filepath = os.path.join(os.path.dirname(__file__), "README.md")
+    with open(readme_filepath) as f:
+        return f.read()
+
+
+setup(
+    name="hydra-optuna-sweeper",
+    version="0.0.1",
```
[the remark this answers is not in the document]
Comment (omry) [reply plugins/hydra_optuna_sweeper/setup.py:16]: Or actually, since we are expecting the change to the config structure:
0.9.0-rc1

**Comment by omry** [comment]
Okay, merged to the 1.0_branch.
Try to release from there. If you run into difficulties let me know.

**Comment by toshihikoyanase** [comment]
@omry Thank you for your instruction. I have released Optuna packages, and I think I can release the plugin by myself. I'll make a release from `1.0_branch`.

**Comment by omry** [comment]
What do you mean, have you released the hydra-optuna-sweeper package to pypi yet? I am not seeing it there.
Don't  forget to make me an owner as well once you do.

**Comment by toshihikoyanase** [comment]
What I wanted to say was that I have released some other libraries and I know how to release packages at pypi.

Now, I released the hydra-optuna-sweeper package [here](https://pypi.org/project/hydra-optuna-sweeper/) and invited you as an owner. Please check the invitation.

**Comment by omry** [comment]
Yup, got it. looking good!
Can you send a PR bringing the [plugin page](https://hydra.cc/docs/next/plugins/optuna_sweeper) to section in part with the rest?

**Comment by toshihikoyanase** [comment]
I'm sorry but I didn't get your point well. Should I add missing badges such as `pypi` and `license` like the other sweepers as follows?

image

**Comment by omry** [comment]
Yup, this is exactly what I mean!
send a PR against master to update it.

**Comment by toshihikoyanase** [comment]
Thank you for your reply. I submitted #1172, so please take a look!

**Comment by omry** [comment]
Awesome.

**Pull request merged**